src/fidesops/service/connectors/query_config.py

```diff
@@ -256,6 +256,76 @@ def generate_update_stmt(
         returns None"""
 
 
+class ManualQueryConfig(QueryConfig[Executable]):
+    def generate_query(
+        self, input_data: Dict[str, List[Any]], policy: Optional[Policy]
+    ) -> Optional[ManualAction]:
+        """Describe the details needed to manually retrieve data from the
+        current collection.
+
+        Example:
+        {
+            "step": "access",
+            "collection": "manual_dataset:manual_collection",
+            "action_needed": [
+                {
+                    "locators": {'email': "customer-1@example.com"},
+                    "get": ["id", "box_id"]
+                    "update":  {}
+                }
+            ]
+        }
+
+        """
+
+        locators: Dict[str, Any] = self.node.typed_filtered_values(input_data)
+        get: List[str] = [
+            field_path.string_path
+            for field_path in self.node.node.collection.top_level_field_dict
+        ]
+
+        if get and locators:
+            return ManualAction(locators=locators, get=get, update=None)
+        return None
+
+    def query_to_str(self, t: T, input_data: Dict[str, List[Any]]) -> str:
+        """Convert query to string"""
+
+    def dry_run_query(self) -> Optional[str]:
```

Comment:
This doesn't do anything? It says it returns an `Optional[str]`, but if it doesn't do anything it will always be `None`

Reply:
thanks, I should have returned None since I wasn't planning on supporting this required method, but thinking about it, there's no reason I can't show the same raw components that I've built here if someone wants to do a dry run of their privacy request.